src/bodywork/git.py

```diff
@@ -44,102 +51,166 @@ def download_project_code_from_repo(
     """
     log = bodywork_log_factory()
     try:
-        run(['git', '--version'], check=True, stdout=DEVNULL)
+        run(["git", "--version"], check=True, stdout=DEVNULL)
     except CalledProcessError:
-        raise RuntimeError('git is not available')
+        raise RuntimeError("git is not available")
     try:
-        if (get_connection_protocol(url) is ConnectionPrototcol.SSH
-                and get_remote_repo_host(url) is GitRepoHost.GITHUB):
-            setup_ssh_for_github()
-        elif SSH_GITHUB_KEY_ENV_VAR not in os.environ:
-            log.warning('Not configured for use with private GitHub repos')
+        if get_connection_protocol(url) is ConnectionProtocol.SSH:
+            hostname = urlparse(f"ssh://{url}").hostname
+            if hostname is not None and hostname != "":
```

Comment:
You could abbreviate this line with `if hostname:`.

You were probably not aware, that `None` is cast to `False`, likewise for `""` or any empty list/container.

The section on `truthiness` in [this post](https://blog.jrheard.com/truthiness-and-short-circuit-evaluation-in-python) is worth reading, but the whole article is worth some time at some point.

Reply:
Noted and corrected!